examples/line-by-line.rs

```diff
@@ -0,0 +1,52 @@
+extern crate tokio_stdin_stdout;
+extern crate tokio;
+extern crate tokio_core;
+extern crate tokio_io;
+
+use tokio::prelude::future::ok;
+use tokio::prelude::Future;
+use tokio_io::io::write_all as tokio_write_all;
+use tokio_io::io::read_exact as tokio_read_exact;
+
+use std::io::ErrorKind;
+
+fn async_op(input: String) -> Box<Future<Item = String, Error = ()> + Send> {
+  Box::new(ok(input.to_ascii_uppercase()))
+}
+
+fn main() {
+  let mut stdin = tokio_stdin_stdout::stdin(0);
+  let stdout = tokio_stdin_stdout::stdout(0).make_sendable();
+
+  let mut current_line = String::from("");
+
+  loop {
+    let buf = vec![0; 1];
```

Comment:
One byte at a time looks inefficient. Maybe there should be `tokio-codec` or something?